resize-observer/observe.html

```diff
@@ -886,19 +929,19 @@
       notify: entries => {
         assert_equals(entries.length, 1, "1 pending notification");
         assert_equals(entries[0].target, t, "target is t");
-        assert_equals(entries[0].contentRect.width, 50, "target width");
-        assert_equals(entries[0].contentRect.height, 100, "target height");
-        assert_equals(entries[0].contentBoxSize[0].inlineSize, 50,
+        assert_rounded_equals(entries[0].contentRect.width, 50, "target width");
```

Comment:
I do not think you need to round sizes for comparasance of contentBoxSize, borderBoxSize, or contentRect if you
choose integer sizes as observed DIV dimensions.

The rounding is necessary for devicePixelContentBoxSize because of floating point math.

Reply:
Are you sure this true? My evidence that the round is needed,

1. Go https://wpt.live/resize-observer/observe.html in Chrome
2. Press Ctrl/Cmd Minus until your devicePixelRatio is 1.333..... (for me on a MacBookPro that's setting the zoom level to 67%
3. Reload the page

For me, 13 tests fail. Where as, if I run it at zoom = 100% then only 2 fail.

So it seems like either (a) the test is wrong or (b) Chrome is wrong. I made the assumption that Chrome is not wrong and so the round is needed.

Reply:
Let me add, I don't know how values are derived but it's certainly valid for the results to be non-integers. I can only guess that whatever chrome is doing, at the time it gets to the ResizeObserverEvent it has to compute a value and it can't round to an integer as that would be incorrect. The computed value is close the to expected value of this test but off by a little. I could change the test to instead of `assert_rounded_equals` to `assert_almost_equals` where it tests `Math.abs(a - b) < Epsilon` but I'm not sure what a good value for Epsilon would be. I guess I could derive it empirically. Checking it looks like an epsilon of 0.04 would pass or 0.032

Reply:
Zoom does make sizes in the test fractional, and you would need the rounding if this page was using zoom.
But this page does not have zoom. There are many wpt tests that would fail if you ran them zoomed.

That is why I do not think that rounding is appropriate.

Reply:
The same problem happens with fractional devicePixelRatio. My Windows PC has a devicePixelRatio of 1.33

Plenty of devices have non-integer devicePixelRatios

https://www.mydevice.io/#compare-devices

Are you saying the tests are require to work only on certain devices?

Reply:
Wow. I did not expect to see 13 failures there. 

I just ran this my PC, I have two monitors, one with dpr of 1.25, and another with 1.5. 
On both of those, only test18 and test19 fail. 

18 and 19 are clearly broken, because they do assume dpr of 1. device_pixel_content_size approximates physical pixels, and those are dpr dependent. 

How are those other tests failing on your machine?

test18 and 19 assert that they are testing:
- test18: an observation is fired when device-pixel-content-box is being observed
- test19: an observation is fired when device-pixel-content-box is being observed and zoom change

These two do not need to assert any sizes, just the fact that observation is fired accomplishes test's intent.

Reply:
> How are those other tests failing on your machine?

because my dpr is 1.333

| test | result |
|------|--------|
| test0: simple observation | assert_equals: target width expected 5 but got 4.984375 |
| test5: observe img | assert_equals: expected 15.5 but got 15.484375 |
| test8: simple content-box observation | assert_equals: target width expected 100 but got 99.984375 |
| test9: simple content-box observation but keep border-box size unchanged | assert_equals: target width expected 100 but got 99.984375 |
| test10: simple border-box observation  assert_equals: target width expected 100 but got 99.984375 |
| test11: simple observation with vertical writing mode | assert_equals: target width expected 50 but got 49.984375 |
| test12: no observation is fired after the change of writing mode when box's specified size comes from logical size properties. | assert_equals: target width expected 50 but got 49.984375 |
| test13: an observation is fired after the change of writing mode when box's specified size comes from physical size properties. | assert_equals: target width expected 50 but got 49.984375 |
| test14: observe the same target but using a different box should override the previous one | assert_equals: target width expected 100 but got 99.984375 |
| test15: an observation is fired with box dimensions 0 when element's display property is set to inline | assert_equals: target width expected 50 but got 49.984375 |
| test17: Box sizing snd Resize Observer notifications | assert_equals: target width expected 40 but got 40.03125 |
| test18: an observation is fired when device-pixel-content-box is being observed | assert_equals: target width expected 50 but got 49.984375 |
| test18: an observation is fired when device-pixel-content-box is being observed | assert_equals: target width expected 50 but got 49.984375 |
| test19: an observation is fired when device-pixel-content-box is being observed and zoom change | assert_equals: target device-pixel-content-box inline size expected 150 but got 200 |

Reply:
Those failures look like a Chrome bug to me. 100px wide div should report a size of 100px for all devices.
The test is correct, the Chrome implementation is buggy. File a bug with Chrome? 

test18 and 19 are bad, they will fail on all dpr != 1